src/core/jsmemops.h

```diff
@@ -0,0 +1,49 @@
+// Macros to access memory from Javascript
+
+#define DEREF_U8(addr, offset) HEAPU8[addr + offset]
+#define DEREF_I8(addr, offset) HEAP8[addr + offset]
+
+#define DEREF_U16(addr, offset) HEAPU16[(addr >> 1) + offset]
+#define DEREF_I16(addr, offset) HEAP16[(addr >> 1) + offset]
+
+#define DEREF_U32(addr, offset) HEAPU32[(addr >> 2) + offset]
+#define DEREF_I32(addr, offset) HEAP32[(addr >> 2) + offset]
+
+#define DEREF_F32(addr, offset) HEAPF32[(addr >> 2) + offset]
+#define DEREF_F64(addr, offset) HEAPF64[(addr >> 3) + offset]
+
+#if WASM_BIGINT
+// We have HEAPU64 / HEAPI64 in this case.
+#define DEREF_U64(addr, offset) HEAPU64[(addr >> 3) + offset]
+#define DEREF_I64(addr, offset) HEAP64[(addr >> 3) + offset]
+
+#define LOAD_U64(addr, offset) DEREF_U64(addr, offset)
+#define LOAD_I64(addr, offset) DEREF_I64(addr, offset)
+
+#define STORE_U64(addr, offset, val) (DEREF_U64(addr, offset) = val)
+#define STORE_I64(addr, offset, val) (DEREF_U64(addr, offset) = val)
+
+#else
+
+// No BigUint64Array, have to manually split / join lower and upper byte
+//
+#define BIGINT_LOWER(x) (Number((x)&BigInt(0xffffffff)) | 0)
+#define BIGINT_UPPER(x) (Number((x) >> BigInt(32)) | 0)
+#define UBIGINT_FROM_PAIR(lower, upper)                                        \
+  (BigInt(lower) | (BigInt(upper) << BigInt(32)))
+
+#define IBIGINT_FROM_PAIR(lower, upper)                                        \
+  (BigInt(lower) | (BigInt(upper - 2 * (upper & 0x80000000)) << BigInt(32)))
```

Comment:
Might be good to remind that in this case BigInt is likely `Number`, right?

Reply:
Well `WASM_BIGINT` is not about whether the browser supports `BigInt` (that may be assumed in all cases), it's about whether it supports `BigInt64Array`.